### PR TITLE
[WIP] - Linting

### PR DESCRIPTION
**Pull request by JosephMontoya-TRI**
Adds a GH action for linting and addresses issues the linter brings up.

Notes:
* I've gotten all of the pycodestyle issues done I believe
* flake8 is complaining about a few issues related to defining variables in logic control flows "out of order" so to speak.
* I haven't begun to address the docstring issues yet, but will before merging.

### CODE REVIEW

**Comment by JosephMontoya-TRI** [comment]
Oh @moorepatrick the reason flake8 is run twice is because on the second time it reports issues with high complexity but doesn't throw errors, which we don't necessarily want to be a stringent requirement, but will want to be reported.

File: beep/utils/events.py [hunk 1]
```diff
@@ -58,34 +59,35 @@ class KinesisEvents:
...
         self.service = service
         self.mode = mode
 
-        if self.mode == 'run':
+        if self.mode == "run":
             for i in range(MAX_RETRIES):
                 try:
-                    self.stream = get_secret(config[ENVIRONMENT]['kinesis']['stream'])['streamName']
+                    self.stream = get_secret(config[ENVIRONMENT]["kinesis"]["stream"])[
+                        "streamName"
```
[the remark this answers is not in the document]
Comment (JosephMontoya-TRI) [reply beep/utils/events.py]: Agreed, split up the secret read and stream assignment to be more readable.

File: .github/workflows/beep-lint.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.8]
```
[the remark this answers is not in the document]
Comment (JosephMontoya-TRI) [reply .github/workflows/beep-lint.yml]: Agreed.  We'll eventually likely want to ensure everything is 3.8 compatible, but for now symmetry makes more sense.

File: .github/workflows/beep-lint.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Linting
+
+on: [push, pull_request]
+
+jobs:
+  build:
```
[the remark this answers is not in the document]
Comment (JosephMontoya-TRI) [reply .github/workflows/beep-lint.yml]: Yep

**Comment by JosephMontoya-TRI** [comment]
So the pycodestyle and flake8 tests are passing now, but I don't think I'm the best person to fix the docstring issues.  We could punt and disable pydocstyle linting for now to be added later, or we could wait.

**Comment by moorepatrick** [comment]
> So the pycodestyle and flake8 tests are passing now, but I don't think I'm the best person to fix the docstring issues. We could punt and disable pydocstyle linting for now to be added later, or we could wait.

I think it's ok to punt on `pydocstyle` for now.

File: beep/structure.py [hunk 4]
```diff
@@ -845,30 +1062,63 @@ def determine_structuring_parameters(self, v_range=None, resolution=1000,
...
+                "diagnostic_start_cycle",
+                "diagnostic_interval",
+            }.issubset(run_parameter.columns):
+                if run_parameter["diagnostic_type"].iloc[0] == "HPPC+RPT":
+                    hppc_rpt = ["reset", "hppc", "rpt_0.2C", "rpt_1C", "rpt_2C"]
                     hppc_rpt_len = 5
-                    diagnostic_starts_at = [1, 1 + run_parameter['diagnostic_start_cycle'].iloc[0] + 1 * hppc_rpt_len]
+                    diagnostic_starts_at = [
+                        1,
+                        1
```
Comment (patrickherring-TRI) [inline beep/structure.py:1090]: This seems like it's a little hard to read.. Does flake8 complain if the addition is all in a single line?

Comment (JosephMontoya-TRI) [reply beep/structure.py]: updated these two lines, flake8 wasn't complaining, but black reformatted this way.

File: beep/features/featurizer_helpers.py [hunk 5]
```diff
@@ -137,30 +169,40 @@ def generate_model(spec):
     return composite_model, params
 
 
-def update_spec_from_peaks(spec, model_indices, peak_voltages, peak_dQdVs, peak_widths=(10,), **kwargs):
-    x = spec['x']
-    y = spec['y']
+def update_spec_from_peaks(
+    spec, model_indices, peak_voltages, peak_dQdVs, peak_widths=(10,), basis_func=None,
```
[the remark this answers is not in the document]
Comment (JosephMontoya-TRI) [reply beep/features/featurizer_helpers.py:173]: It's either line length constraints on black or I may have revised this by hand this after the last merge.

**Pull request merged**